ruby/publish/action.yml

```
@@ -0,0 +1,156 @@
+  gem_name:
+    description: The name (sans extension) of the gemspec file (e.g. "mongo")
+    required: true
+  product_name:
+    description: The name of the product being published (e.g. "Ruby Driver")
+    required: true
+  product_id:
+    description: The identifier of the product being published (e.g. "mongo-ruby-driver")
+    required: true
+  release_message_template:
+    description: The template for the release message. Use "{0}" in the text to refer to the current version.
+    required: true
+  silk_asset_group:
+    description: The Silk asset group for the project
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: "Create temporary app token"
+      uses: actions/create-github-app-token@v1
+      id: app-token
+      with:
+        app-id: ${{ inputs.app_id }}
+        private-key: ${{ inputs.app_private_key }}
+
+    - name: "Store GitHub token in environment"
+      shell: bash
+      run: echo "GH_TOKEN=${{ steps.app-token.outputs.token }}" >> "$GITHUB_ENV"
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ env.GH_TOKEN }}
+
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '3.2'
+        bundler-cache: true
+
+    - name: Get the release version
+      shell: bash
+      run: echo "RELEASE_VERSION=$(bundle exec rake version)" >> "$GITHUB_ENV"
+
+    - name: Setup GitHub tooling for DBX Drivers
+      uses: mongodb-labs/drivers-github-tools/setup@v2
+      with:
+        aws_role_arn: ${{ inputs.aws_role_arn }}
+        aws_region_name: ${{ inputs.aws_region_name }}
+        aws_secret_id: ${{ inputs.aws_secret_id }}
+
+    - name: Set output gem file name
+      shell: bash
+      run: |
+        echo "GEM_FILE_NAME=${{ inputs.gem_name }}-${{ env.RELEASE_VERSION }}.gem" >> "$GITHUB_ENV"
+
+    - name: Build the gem
+      shell: bash
+      run: |
+        gem build --output=${{ env.GEM_FILE_NAME }} ${{ inputs.gem_name }}.gemspec
+
+    - name: Sign the gem
+      uses: mongodb-labs/drivers-github-tools/gpg-sign@v2
+      with:
+        filenames: '${{ env.GEM_FILE_NAME }}'
+
+    - name: Generate authorized publication Documentation
+      uses: mongodb-labs/drivers-github-tools/authorized-pub@v2
+      with:
+        product_name: ${{ inputs.product_name }}
+        release_version: ${{ env.RELEASE_VERSION }}
+        filenames: ${{ env.GEM_FILE_NAME }}
+        token: ${{ env.GH_TOKEN }}
+
+    - name: Download SBOM file from Silk
+      uses: mongodb-labs/drivers-github-tools/sbom@v2
+      with:
+        silk_asset_group: ${{ inputs.silk_asset_group }}
+
+    - name: Generate Sarif Report
+      uses: mongodb-labs/drivers-github-tools/code-scanning-export@v2
+      with:
+        output-file: ${{ env.S3_ASSETS }}/code-scanning-alerts.json
+
+    - name: Generate Compliance Report
+      uses: mongodb-labs/drivers-github-tools/compliance-report@v2
+      with:
+        token: ${{ env.GH_TOKEN }}
```

The `full-report` action handles these for steps for you:

```suggestion
    - name: "Generate SSDLC Reports"
      uses: mongodb-labs/drivers-github-tools/full-report@v2
      with:
        product_name: ${{ inputs.product_name }}
        release_version: ${{ env.RELEASE_VERSION }}
        dist_filenames: ${{ env.GEM_FILE_NAME }}
        silk_asset_group: ${{ inputs.silk_asset_group }}
```

This action also uses the `GH_TOKEN` env variable by default.